great_expectations/expectations/expectation.py

```diff
@@ -1166,11 +1151,22 @@ def get_validation_dependencies(
             metric_configurations={}, result_format=result_format
         )
 
+    def _get_default_value(self, key: str) -> Any:
+        """Get default value for the model if it is not required.
+        NOTE: For reasons I'm unclear on, we can get back either a FieldInfo or a ModelField.
+        """
+        field = self.__fields__.get(key)
+
+        if isinstance(field, pydantic.fields.ModelField):
+            return field.default if not field.required else None
+        else:
+            return None
```

Comment:
Will we ever call this code with a value that doesn't exist? I'd rather we fail than possibly pass down `None` when we shouldn't

Reply:
We definitely shouldn't, but that case was implicitly handled like this before, and I wanted to avoid behavior changes. If you think I should throw a `raise` in here, I'm happy to do it if you think that's better.

Reply:
Hmm I agree based on that reasoning but maybe we log before returning `None` just to provide context around the miss if it happens?